sdk/nodejs/cmd/pulumi-language-nodejs/main.go

```diff
@@ -740,7 +767,34 @@ func (host *nodeLanguageHost) execNodejs(ctx context.Context, req *pulumirpc.Run
 		opentracing.Tag{Key: "args", Value: nodeargs})
 	defer tracingSpan.Finish()
 
-	if err := cmd.Run(); err != nil {
+	run := func() error {
+		err := cmd.Start()
+		if err != nil {
+			return err
+		}
+		if req.GetAttachDebugger() {
+			debugConfig, err := structpb.NewStruct(map[string]interface{}{
+				"name":             "Pulumi: Program (Node.js)",
+				"type":             "node",
+				"request":          "attach",
+				"processId":        cmd.Process.Pid,
+				"continueOnAttach": true,
+				"skipFiles":        []interface{}{"<node_internals>/**"},
+			})
+			if err != nil {
+				return err
+			}
+			_, err = engineClient.StartDebugging(ctx, &pulumirpc.StartDebuggingRequest{
```

Comment:
Do we not need to wait for node to have started here?

Trying `--inspect-brk` locally, and it's really fast, so maybe not? I think all the imports for the program are done after the program continues running, so this might work even with very large SDKs.

If we do have to wait, we can probably try to dial the TCP port on which the node debugger listens, and if that's available, we know we're ready.

Reply:
Hmm it's probably not a bad idea to guard against slow node starts here.  Let me implement this.

Reply:
Although, on second thought, since we're using the process ID here, we're sure the process has started by the time we get here.  So it should always be safe to give this information to the IDE since it can always attach to the process, and we shouldn't need to wait for the TCP port.